x-pack/plugins/enterprise_search/server/lib/check_access.ts

```diff
@@ -98,6 +98,6 @@ export const checkAccess = async ({
 
   // When enterpriseSearch.host is defined in kibana.yml,
   // make a HTTP call which returns product access
-  const { access } = (await callEnterpriseSearchConfigAPI({ request, config, log })) || {};
-  return access || DENY_ALL_PLUGINS;
+  const response = (await callEnterpriseSearchConfigAPI({ request, config, log })) || {};
+  return 'access' in response ? response.access || DENY_ALL_PLUGINS : DENY_ALL_PLUGINS;
```

Comment:
Previous version of this code and this is eventually the same. Unless I miss something of course.

If response from API is truthy and has no `access` property inside, on the next line `access` would be defined as `undefined`. Therefore `access || DENY_ALL_PLUGINS` would return `DENY_ALL_PLUGINS`

If access exists but not truthy, that would work the same way still.

Reply:
I'm not an expert in Typescript, but the linter told me that things will break trying too unpack `const { access }`, because the return type `ResponseError` doesn't have the `access` property. You think it would work?

Reply:
Apparently Kibana tests agree with linter.

https://buildkite.com/elastic/kibana-pull-request/builds/42640#43c02556-a378-45d0-a456-05ec8ceecead

I'm reverting back to my version.